### ISSUE TITLE
T10 elemental shaman 4 piece bonus doesn't work properly

### ISSUE THREAD

**Issue by pgcristyan** [dalaranwow/dalaran-wow#5922]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
The 4 piece T10 bonus is: "Your Lava Burst spell causes your Flame Shock effect on the target to deal at least two additional periodic damage ticks before expiring." The issue on dalaran wow is that Lava Burst always adds just 2 ticks to flame shock, regardless of haste values.
**Expected behaviour**: Tell us what should happen instead
What should happen, is lava burst should give you **at least** two additional ticks. What this means, is that it should give you the amount of ticks closest to an added 6 seconds to flame shock. So, depending on haste values, if your flame shock ticks every 3 seconds (0 haste), then it should add 2 ticks. If your haste lowers the tick time down to 2 seconds, it should add 3 ticks. If your haste lowers the tick time down to 1.5 seconds, it will add 4 ticks. Even when taken literally, the bonus says **at least** two additional ticks, but on our server we get 2 ticks all the time.
**Steps to reproduce the problem**:

1. Equip 4-piece T10 on elemental shaman.
2. Use hasting abilities that get your total flame shock duration below 12 seconds (as flame shock has baseline 6 ticks, this means you get your flame shock ticks every 2 seconds or less) - I used Elemental mastery and Heroism at the same time.
3. Cast Flame shock on target and then immediately Lava Burst. This should add 3 additional Flame shock ticks(total of 9 ticks), but actually it only adds 2(total of 8 ticks).

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

Comments on the bonus: https://www.wowhead.com/spell=70817/item-shaman-t10-elemental-4p-bonus#comments
Patch changes: https://wow.gamepedia.com/Frost_Witch%27s_Regalia
The official patch changes from when the changes was made say that the bonus now accounts for haste:
"This bonus has been slightly adjusted to account for the fact that haste now modifies Flame Shock's periodic damage ticks. The bonus now makes the shaman's Lava Burst cause Flame Shock to tick at least two additional times before expiring."
In-depth analysis of the change when it went live by an elemental shamy: https://totemz.wordpress.com/2010/03/13/patch-3-3-3-and-elemental-changes/

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

Flame shock ID: https://wotlkdb.com/?spell=49233
Lava Burst ID: https://wotlkdb.com/?spell=60043
Tier 10 4-piece bonus ID: https://wotlkdb.com/?spell=70817

**Include Screenshots from the issue if necessary**
